addons/mail/models/mail_message.py

```diff
@@ -1020,20 +1020,28 @@ def message_fetch(self, domain, limit=20, moderated_channel_ids=None):
               pending moderation;
             :returns list(dict).
         """
-        messages = self.search(domain, limit=limit)
         if moderated_channel_ids:
             # Split load moderated and regular messages, as the ORed domain can
             # cause performance issues on large databases.
```

Comment:
When I see this comment I think reintroducing OR is probably a bad idea.
I think we well need to live with this being broken until we do all our other changes, and in particular when we are finally able to get rid of channel_ids and ensure that messages can be in only one thread at once. This is the only way to get rid of stupid "OR" conditions that are really bad for performance.
(alternative if we do want to keep channel_ids somehow is that moderated messages have to have it set in the first place, and be distinguished from other messages by a proper flag, which even already exists).